app/workers/calculate_bill_worker.rb

```diff
@@ -0,0 +1,36 @@
+class CalculateBillWorker
+  AVERAGE_METER_READING = 20
+  include Sidekiq::Worker
+
+  def perform(*_args)
+    BillingContract.all.each do |billing_contract|
+      delta_meter_reading = get_delta_reading(billing_contract)
+      bill_value = calculate_bill(delta_meter_reading, billing_contract)
+      bill_value = edit_value_depending_payments(bill_value, billing_contract)
+      byebug
+      FactoryBot.create(:bill, amount: bill_value, date: Time.now.utc, billing_contract: billing_contract,
+                               meter_reading: meter_reading)
+    end
+  end
+
+  def get_delta_reading(billing_contract)
+    meter_reading = billing_contract.meter_readings.created_between(billing_contract.last_bill.date, Time.now.utc)
+                                    .ordered_by_date.first&.value
+    last_meter_reading = billing_contract.last_bill&.meter_reading
+    last_meter_reading ? meter_reading - last_meter_reading : meter_reading
+  end
+
+  def edit_value_depending_payments(value, contract)
+    payments = contract.payments.created_between(contract.last_bill.date, Time.now.utc)
```

Comment:
if you use payments variable only once, no need to create it

Reply:
refactor